SIPS/sip-29.md

```diff
@@ -0,0 +1,195 @@
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+1. In this document, all definitions are written in TypeScript.
+
+2. Any time an asset needs to be identified, it MUST use the [CAIP-19][caip-19]
+representation.
+
+### Snap Assets API
+
+Two methods are defined in the Snap Assets API:
+
+#### Get Token Description
+
+```typescript
+// Represents a token unit.
+type TokenUnit = {
+    // Human-friendly name of the token unit.
+    name: string;
+
+    // Ticker of the token unit.
+    ticker: string;
+
+    // Number of decimals of the token unit.
+    decimals: number;
+};
+
+// Token description.
+type TokenDescription = {
+    // Human-friendly name of the token.
+    name: string;
+
+    // Ticker of the token.
+    ticker: string;
+
+    // Whether the token is native to the chain.
+    isNative: boolean;
+
+    // Base64 representation of the token icon.
+    iconBase64: string;
+
+    // List of token units.
+    units: TokenUnit[];
+};
+
+// Returns the description of a non-fungible token. This description can then
+// be used by the client to display relevant information about the token.
+//
+// @example
+// ```typescript
+// const tokenDescription = await getTokenDescription('eip155:1/slip44:60');
+//
+// // Returns:
+// // {
+// //     name: 'Ether',
+// //     ticker: 'ETH',
+// //     isNative: true,
+// //     iconBase64: 'data:image/png;base64,...',
+// //     units: [
+// //         {
+// //             name: 'Ether',
+// //             ticker: 'ETH',
+// //             decimals: 18
+// //         },
+// //         {
+// //             name: 'Gwei',
+// //             ticker: 'Gwei',
+// //             decimals: 9
+// //         },
+// //         {
+// //             name: 'wei',
+// //             ticker: 'wei',
+// //             decimals: 0
+// //         }
+// //     ]
+// // }
+// ```
+function getTokenDescription(token: Caip19AssetType): TokenDescription;
+```
+
+#### Get Token Conversion Rate
+
+```typescript
+type TokenConversionRate = {
+    // The rate of conversion from the source token to the target token. It
+    // means that 1 unit of the `from` token should be converted to this amount
+    // of the `to` token.
+    rate: string;
+
+    // The UNIX timestamp of when the conversion rate was last updated.
+    conversionTime: number;
+
+    // The UNIX timestamp of when the conversion rate will expire.
+    expirationTime: number;
+};
+
+// Returns the conversion rate between two assets (tokens or fiat).
+//
+// @example
+// ```typescript
+// const conversionRate = await getTokenConversionRate(
+//   'eip155:1/slip44:60',
+//   'eip155:1/erc20:0x6b175474e89094c44da98b954eedeac495271d0f'
+// );
+//
+// // Returns:
+// // {
+// //     rate: '3906.38',
+// //     conversionTime: 1733389786,
+// //     expirationTime: 1733389816,
+// // }
+// ```
+function getTokenConversionRate(
+    from: Caip19AssetType,
+    to: Caip19AssetType
+): TokenConversionRate;
+```
+
+### Fiat currency representation
```

Comment:
We should create a CAIP for this

Reply:
This is a good point. This is technically CAIP-19, but with unregistered namespaces.

I started a [discussion](https://github.com/ChainAgnostic/CAIPs/discussions/338) on the CAIPs repo, let's see what other think of it.